mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/voice/VoiceInputCombinator.kt

```diff
@@ -20,36 +27,75 @@ object VoiceInputCombinator {
    */
   // TODO : Internationalization
   private val rank =
+  // Try to transformed into a our chess word
+  combine(
+              convertToken(ChessSpeechEnglishDictionary.chessPieces),
+              token().map { token ->
+                rulesSet
+                    .map { chessRule ->
+                      // Apply rule
+                      val res = chessRule.rule(token)
+                      res
+                    }
+                    .firstOrNull { res ->
+                      // Get the 1st value that have been transformed
+                      res != null
+                    }
+              })
+          .map {
+            when (it) {
+              "king" -> Rank.King
+              "queen" -> Rank.Queen
+              "rook" -> Rank.Rook
+              "bishop" -> Rank.Bishop
+              "knight" -> Rank.Knight
+              "pawn" -> Rank.Pawn
+              else -> null
+            }
+          }
+
+  val column =
+      convertToken(ChessSpeechEnglishDictionary.letters)
+          .filter { it in 'a'..'h' }
+          .filterNotNull()
+          .map { it!! - 'a' }
```

Comment:
With the filterNotNull, the `!!` is not needed anymore.
```suggestion
          .map { it - 'a' }
```

Reply:
Fix https://github.com/epfl-SDP/android/pull/311/commits/3d47c6bbc497a38208cc8894e49b95b6e73f319a

Thx, I forgot to delete `!!`